src/renderer/src/components/Contact/ContactList.tsx

```diff
@@ -1,34 +1,50 @@
 import { Fragment, useMemo, useState } from 'react'
 import { useTranslation } from 'react-i18next'
-import { TbCheck } from 'react-icons/tb'
-import { IContactState } from '@renderer/@types/store'
+import { TbCheck, TbChevronUp } from 'react-icons/tb'
+import { IContactState, TContactAddress } from '@renderer/@types/store'
 import { SearchInput } from '@renderer/components/SearchInput'
+import { StringHelper } from '@renderer/helpers/StringHelper'
 import { StyleHelper } from '@renderer/helpers/StyleHelper'
 import { useContactsSelector } from '@renderer/hooks/useContactSelector'
 import { cloneDeep } from 'lodash'
 
+import { BlockchainIcon } from '../BlockchainIcon'
 import { Separator } from '../Separator'
 
 type TProps = {
-  onContactSelected: (contact: IContactState | null) => void
+  onContactSelected?: (contact: IContactState | null) => void
+  onAddressSelected?: (address: TContactAddress | null) => void
   selectFirst: boolean
   showSelectedIcon: boolean
   children?: React.ReactNode
 }
 
-export const ContactList = ({ onContactSelected, selectFirst, showSelectedIcon, children }: TProps) => {
+export const ContactList = ({
+  onContactSelected,
+  onAddressSelected,
+  selectFirst,
+  showSelectedIcon,
+  children,
+}: TProps) => {
   const { t: contactT } = useTranslation('components', { keyPrefix: 'contacts' })
   const { contacts } = useContactsSelector()
   const [search, setSearch] = useState<string | null>(null)
   const [selectedContact, setSelectedContact] = useState<IContactState | null>(selectFirst ? contacts[0] || null : null)
+  const [selectedAddress, setSelectedAddress] = useState<TContactAddress | null>(null)
 
-  onContactSelected(selectedContact)
+  onContactSelected && onContactSelected(selectedContact)
+  onAddressSelected && onAddressSelected(selectedAddress)
```

Comment:
This will call onContactSelected and onAddressSelected for every render of this component. It would be better if it was only called when the selection actually happens.

```suggestion
 const handleContactSelected = (contact: IContactState) => {
    onContactSelected(contact)
    ...more code
  }
  
   const handleAddressSelected = (address: TContactAddress) => {
    onAddressSelected(address)
    ...more code
  }

```